pyccel/parser/semantic.py

```diff
@@ -506,11 +511,25 @@ def remove_variable(self, name):
     def get_header(self, name):
         """."""
         container = self.namespace
+        headers = []
         while container:
             if name in container.headers:
-                return container.headers[name]
+                if isinstance(container.headers[name], list):
+                    headers += container.headers[name]
```

Comment:
You could also use `headers.extend(container.headers[name])`. I tried to search to find out if there's a speed difference but I couldn't find anything so it's up to you which you prefer

Reply:
I think the magic method `__iadd__` of a list just calls `extend`, so there should be no difference at all. I personally like to use `+` and `+=` for list concatenation.

Reply:
I will change it